index.js

```diff
@@ -1986,14 +1988,31 @@ export function generate(options) {
     let rightSize = false;
     let wordUsed;
     while (!rightSize) {
-      wordUsed = generateRandomWord();
+      // Generate a random word with the provided prefix (if any)
+      wordUsed = generateRandomWord(prefix);
       rightSize = wordUsed.length <= max && wordUsed.length >= min;
     }
     return wordUsed;
   }
 
-  function generateRandomWord() {
-    return wordList[randInt(wordList.length)];
+  // Function to generate a random word with an optional prefix
+  function generateRandomWord(prefix = null) {
+    let result = null;
+
+    // Filter the wordList to find words starting with the specified prefix
+    if (prefix) {
+      const filteredList = wordList.filter((word) =>
+        word.startsWith(prefix)
+      );
+      if (filteredList.length > 0) {
```

Comment:
Throw an exception (`throw new Error('Unavailable')`) if no words are available meeting the criteria.